docs/literate/src/files/differentiable_programming_literate.jl

```diff
@@ -0,0 +1,409 @@
+#src # Differentiable programming
+using Test #src
+
+# [Julia and its ecosystem provide some tools for differentiable programming](https://sinews.siam.org/Details-Page/scientific-machine-learning-how-julia-employs-differentiable-programming-to-do-it-best).
+# Trixi.jl is designed to be flexible, extendable, and composable with Julia's growing ecosystem for
+# scientific computing and machine learning. Thus, the ultimate goal is to have fast implementations
+# that allow automatic differentiation (AD) without too much hassle for users. If some parts do not
+# meet these requirements, please feel free to open an issue or propose a fix in a PR.
+
+# In the following, we will walk through some examples demonstrating how to differentiate through
+# Trixi.jl.
+
+
+# ## Forward mode automatic differentiation
+
+# Trixi integrates well with [ForwardDiff.jl](https://github.com/JuliaDiff/ForwardDiff.jl)
+# for forward mode AD.
+
+
+# ### Computing the Jacobian
+# The high-level interface to compute the Jacobian this way is [`jacobian_ad_forward`](@trixi-docs:reference-trixi/#Trixi.jacobian_ad_forward-Tuple{Trixi.AbstractSemidiscretization}).
+
+using Trixi, LinearAlgebra, Plots
+#-
+equations = CompressibleEulerEquations2D(1.4);
+
+solver = DGSEM(3, flux_central);
+
+mesh = TreeMesh((-1.0, -1.0), (1.0, 1.0), initial_refinement_level=2, n_cells_max=10^5);
+
+semi = SemidiscretizationHyperbolic(mesh, equations, initial_condition_density_wave, solver);
+
+J = jacobian_ad_forward(semi);
+
+size(J)
+@test size(J) == (1024, 1024) #src
+#-
+λ = eigvals(J);
+
+scatter(real.(λ), imag.(λ));
+
+3.0e-10 < maximum(real, λ) / maximum(abs, λ) < 8.0e-10
+@test (3.0e-10 < maximum(real, λ) / maximum(abs, λ) < 8.0e-10) == true #src
```

Comment:
```suggestion
maximum(real, λ) / maximum(abs, λ)
@test (3.0e-10 < maximum(real, λ) / maximum(abs, λ) < 8.0e-10) == true #src
```
Is the output of the line `maximum(real, λ) / maximum(abs, λ)` shown online? If so, it's nice to just print the exact value of interest and to test some condition afterwards as you do above. This also applies to other parts

Reply:
Good idea. For now, I just used the existing tutorials.

Reply:
I don't have a strong opinion on this. For me, it would be okay to just show the standard output. Rounding would introduce some additional noise in the code but may make the output more clear. I guess it depends a bit on personal taste and the look an feel of the final output.

Reply:
That's what I thought, too. Maybe let's just start with the pure output and see what it looks like.